lib/api/model/events.dart

```diff
@@ -850,6 +852,68 @@ class UpdateMessageFlagsMessageDetail {
   Map<String, dynamic> toJson() => _$UpdateMessageFlagsMessageDetailToJson(this);
 }
 
+
+/// A Zulip event of type `typing`:
+///   https://zulip.com/api/get-events#typing-start
+///   https://zulip.com/api/get-events#typing-stop
+@JsonSerializable(fieldRename: FieldRename.snake)
+class TypingEvent extends Event {
+  @override
+  @JsonKey(includeToJson: true)
+  String get type => 'typing';
+
+  final TypingOp op;
+  final MessageType messageType;
```

Comment:
> OTOH if the most convenient way to add the logic winds up being separate for each place the enum is used, then this PR might as well stick to this one spot where it's actually needed, and we can handle the others later.

I put the change that affected other events in a prep commit: https://github.com/zulip/zulip-flutter/pull/783/commits/9410969ed75edd1154007811e4278d32e2d2db6b
This might not be the cleanest way to do it, but all we need is annotating the field with the converter.